src/main/java/org/opensearch/security/identity/PluginContextSwitcher.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+package org.opensearch.security.identity;
+
+import java.util.Objects;
+import java.util.concurrent.Callable;
+
+import org.opensearch.identity.PluginSubject;
+
+public class PluginContextSwitcher {
```

Comment:
Its a vehicle for carrying subject that's been assigned to a plugin to the transport layer. Take a look at `SystemIndexPlugin1` where it instantiates class and returns it from `createComponents` so that its injectable to transport actions.